### PR TITLE
Adding and modifying methods for Player effects

### PR DESCRIPTION
**Pull request by Thundermaker300**
* <s>`Player::GetActiveEffects()` (PlayerEffect[]) - Returns an array of active player effects</s>
* <s>`Player::EnableEffects(string[], float, bool)` (bool) - Enables a list of effects for duration, returns true if all successful</s>
* `Player::EffectActive<T>()` (bool) - Returns if given type is active
* `Player::ChangeEffectIntensity<T>(byte)` (void) - Changes the effect intensity
* `Player::ChangeEffectIntensity(string, byte)` (void) - Changes the effect intensity
* `Player::GetEffectIntensity<T>()` (byte)` - Returns effect intensity
* Modified: `Player::DisableAllEffects` to only call disabled on the enabled effects
(Apologies for like my 20th PR in the past week)

### CODE REVIEW

**Comment by Thundermaker300** [comment]
I think that's enough commits lol

File: Exiled.API/Features/Player.cs [hunk 1]
```diff
@@ -1222,23 +1222,58 @@ public void ShowHint(string message, float duration = 3f)
...
+            foreach (KeyValuePair<Type, PlayerEffect> data in ReferenceHub.playerEffectsController.AllEffects)
+            {
+                if (data.Value.Enabled)
+                {
+                    arr.Append(data.Value);
+                }
+            }
+
+            return arr;
+        }
```
Comment (iRebbok) [inline Exiled.API/Features/Player.cs:1240]: Is it really useful? It literally allocates N arrays, where N is the number of enabled effects. I don't find it so useful.

File: Exiled.API/Features/Player.cs [hunk 2]
```diff
@@ -1247,14 +1282,70 @@ public void EnableEffect<T>(float duration = 0f, bool addDurationIfActive = fals
...
+            {
+                bool success = ReferenceHub.playerEffectsController.EnableByString(str, duration, addDurationIfActive);
+                if (!success)
+                {
+                    flag = false;
+                }
+            }
+
+            return flag;
+        }
```
Comment (iRebbok) [inline Exiled.API/Features/Player.cs:1313]: Same as with `GetActiveEffects()`, does anyone need to enable a bunch of events with the same options?

Comment (Thundermaker300) [reply Exiled.API/Features/Player.cs:1313]: I've had a case before where I could've used this but it wasn't available, although I see your point

File: Exiled.API/Features/Player.cs (hunk 1 above)
Comment (Thundermaker300) [reply Exiled.API/Features/Player.cs:1240]: fair enough

File: Exiled.API/Features/Player.cs (hunk 2 above)
Comment (iRebbok) [reply Exiled.API/Features/Player.cs]: I mean, it'd be very useful.. but using types instead of string

**Comment by iRebbok** [comment]
@Thundermaker300 ready?

**Comment by Thundermaker300** [comment]
@iRebbok yeah, I don't think there's anything else necessary for this PR, unless you have ideas

**Pull request merged**